internal/setup/setup.go

```diff
@@ -534,14 +535,34 @@ func parsePackage(baseDir, pkgName, pkgPath string, data []byte) (*Package, erro
 			Scripts: SliceScripts{
 				Mutate: yamlSlice.Mutate,
 			},
+			Essential: map[SliceKey]bool{},
+		}
+		for _, refName := range yamlPkg.Essential {
+			sliceKey, err := ParseSliceKey(refName)
+			if err != nil {
+				return nil, fmt.Errorf("invalid slice reference %q in %s", refName, pkgPath)
+			}
+			if sliceKey.Package == slice.Package && sliceKey.Slice == slice.Name {
+				// Do not add the slice to its own essentials list.
+				continue
+			}
+			if _, ok := slice.Essential[sliceKey]; ok {
+				return nil, fmt.Errorf("cannot reference slice %q twice in essentials for slice %q in %s", refName, slice.String(), pkgPath)
+			}
+			slice.Essential[sliceKey] = true
 		}
-
 		for _, refName := range yamlSlice.Essential {
 			sliceKey, err := ParseSliceKey(refName)
 			if err != nil {
 				return nil, fmt.Errorf("invalid slice reference %q in %s", refName, pkgPath)
 			}
-			slice.Essential = append(slice.Essential, sliceKey)
+			if sliceKey.Package == slice.Package && sliceKey.Slice == slice.Name {
+				return nil, fmt.Errorf("cannot add slice to itself as essential %q in %s", refName, pkgPath)
+			}
+			if _, ok := slice.Essential[sliceKey]; ok {
+				return nil, fmt.Errorf("cannot add %q twice in essentials for slice %q in %s", refName, slice.String(), pkgPath)
```

Comment:
We can use the exact same message above for both of these cases too. It's a redundant essential in all these cases.